corehq/messaging/scheduling/forms.py

```diff
@@ -2951,7 +2989,13 @@ def add_initial_for_custom_metadata(self, result):
             len(self.initial_schedule.custom_metadata) > 0
         ):
             result['capture_custom_metadata_item'] = self.YES
-            for name, value in self.initial_schedule.custom_metadata.items():
+            custom_metadata = self.initial_schedule.custom_metadata
+
+            default_stale_interval_in_days = int(settings.SMS_QUEUE_STALE_MESSAGE_DURATION / 24)
+            sms_stale_after = custom_metadata.pop('sms_stale_after', default_stale_interval_in_days)
+            result['sms_stale_after'] = sms_stale_after
+
+            for name, value in custom_metadata.items():
                 result['custom_metadata_item_name'] = name
                 result['custom_metadata_item_value'] = value
                 # We only capture one item right now, but the framework allows
```

Comment:
I think you would need to get rid of this "break" for this to work now.

Reply:
>  get rid of this "break"
Did not get it. Can you please elaborate?

Reply:
the for loop breaks after the first iteration.

Reply:
It can be left as it is, I am extracting the property above https://github.com/dimagi/commcare-hq/pull/28419/files#diff-594ca011fb9871d019f20b229abad83aR2995